### PR TITLE
Mock model when testing get_document_or_404

### PR DESCRIPTION
**Pull request by kaapstorm**
Avoids hitting CouchDB, and checks specific functionality of code under test.

(FB 146661, from [Test get document](https://github.com/dimagi/commcare-hq/pull/4763#issuecomment-62683742) PR)

### CODE REVIEW

File: corehq/util/tests/test_couch.py [hunk 1]
```diff
@@ -1,22 +1,80 @@
...
+        get_document_or_404 should raise 404 on incorrect domain
+        """
+        with self.assertRaises(Http404):
+            get_document_or_404(MockModel, 'spam', '123')
+
+    def test_get_document_or_404_wrapping_error(self):
+        """
+        get_document_or_404 should raise 404 on WrappingAttributeError
+        """
+        import ipdb; ipdb.set_trace()
```
Comment (dimagimon) [inline corehq/util/tests/test_couch.py]: E702 multiple statements on one line (semicolon)

File: corehq/util/tests/test_couch.py [hunk 2]
```diff
@@ -1,22 +1,80 @@
...
+        with mock_wrap_context():
+            with self.assertRaises(Http404):
+                get_document_or_404(MockModel, 'ham', '123')
+
+    def test_get_document_or_404_success(self):
+        """
+        get_document_or_404 should return a wrapped model on success
+        """
+        import ipdb; ipdb.set_trace()
```
Comment (dimagimon) [inline corehq/util/tests/test_couch.py]: E702 multiple statements on one line (semicolon)

File: corehq/util/tests/test_couch.py (hunk 1 above)
Comment (czue) [inline corehq/util/tests/test_couch.py]: oops

Comment (kaapstorm) [inline corehq/util/tests/test_couch.py]: Oops! :-(

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by kaapstorm** [comment]
(Nice, short, pre-commit hook for checking for i?pdb:  https://gist.github.com/chronossc/0a46e2025154b9a45bce )

**Pull request merged**